### PR TITLE
Generate devilutionx.mpq

### PR DESCRIPTION
**Pull request by AJenbo**
This is probably not the best implementation but it's a working draft at the very least :) Feedback is extremely welcome.

Next step would be:
- adding translations to assets
- removing ttf,
- having platforms that provide there own container prefer that over the MPQ (Android APK)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -675,6 +671,102 @@ if (Gettext_FOUND)
...
+  ui_art/supportw.pcx)
+
+find_program(SMPQ smpq)
+if(SMPQ)
+  file(REMOVE "Packaging/resources/devilutionx.mpq")
+  execute_process(
+	WORKING_DIRECTORY "${DevilutionX_SOURCE_DIR}/Packaging/resources/assets"
+	COMMAND ${SMPQ} -M 1 -C PKWARE -c "../devilutionx.mpq" ${devilutionx_assets})
+  # Copy devilutionx.mpq to the build directory to it works from the build directory
+  file(COPY "Packaging/resources/devilutionx.mpq" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
```
Comment (StephenCWills) [inline CMakeLists.txt:764]: Should `Packaging/resources/devilutionx.mpq` be added to `.gitignore`?

Comment (AJenbo) [reply CMakeLists.txt:764]: We already have `*.mpq` :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -675,6 +671,102 @@ if (Gettext_FOUND)
...
+  ui_art/creditsw.pcx
+  ui_art/hf_titlew.pcx
+  ui_art/mainmenuw.pcx
+  ui_art/supportw.pcx)
+
+find_program(SMPQ smpq)
+if(SMPQ)
+  file(REMOVE "Packaging/resources/devilutionx.mpq")
+  execute_process(
+	WORKING_DIRECTORY "${DevilutionX_SOURCE_DIR}/Packaging/resources/assets"
```
Comment (glebm) [inline CMakeLists.txt:761]: A wild mix of spaces and tabs

**Comment by StephenCWills** [comment]
> `# Copy devilutionx.mpq to the build directory to it works from the build directory` this sentence makes no sense, occurs twice 😄

I'd guess that it's meant to be "Copy devilutionx.mpq so the game works from the build directory".

**Review by glebm — APPROVED** [review]

**Comment by AJenbo** [comment]
Good luck to anyone who does not have smpq available to them :wave:

**Pull request merged**